polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -1752,15 +1757,24 @@ impl<Config: config::Config> XcmExecutor<Config> {
 		context: Option<&XcmContext>,
 	) -> Result<(), XcmError> {
 		let mut failed_deposits = Vec::with_capacity(to_deposit.len());
+		let deposit_result = Ok(());
 
-		let mut deposit_result = Ok(());
 		for asset in to_deposit.assets_iter() {
-			deposit_result = Config::AssetTransactor::deposit_asset(&asset, &beneficiary, context);
+			let deposit_result =
+				Config::AssetTransactor::deposit_asset(&asset, &beneficiary, context);
+
 			// if deposit failed for asset, mark it for retry after depositing the others.
-			if deposit_result.is_err() {
-				failed_deposits.push(asset);
+			if let Err(e) = &deposit_result {
+				// Gracefully skip dust deposits
+				if Self::is_below_minimum_error(e) {
+					Self::do_handle_below_minimum(&asset, &beneficiary, context);
+					continue;
+				} else {
+					failed_deposits.push(asset);
+				}
```

Comment:
We can simply retry all failures in the first go (including dust), and burn it only if it fails on the second try.

This helps support unlikely situations like `[Deposit(dust, Alice), Deposit(enough, Alice)]` -> because of the retry, instead of burning `dust`, we would actually successfully deposit it to Alice on the second try.

```suggestion
			if deposit_result.is_err() {
				failed_deposits.push(asset);
```